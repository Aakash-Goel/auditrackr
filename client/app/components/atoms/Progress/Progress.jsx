import React from 'react';
import { object, oneOf } from 'prop-types';

import withStyles from '@material-ui/core/styles/withStyles';
import LinearProgress from '@material-ui/core/LinearProgress';

import progressStyles from './Progress.style';

/**
 * Type checking - Define prop types
 * @private
 */
const propTypes = {
  classes: object.isRequired,
  color: oneOf(['primary', 'secondary']),
};

/**
 * Define default prop values
 * @private
 */
const defaultProps = {
  color: 'primary',
};

const CustomProgress = ({ ...props }) => {
  const { classes, color, ...rest } = props;

  return (
    <LinearProgress
      {...rest}
      color={color}
      classes={{
        root: `${classes.root} ${classes[`${color}Background`]}`,
        bar: `${classes.bar} ${classes[color]}`,
        bar2Buffer: `${classes[`${color}BufferBackground`]}`,
      }}
    />
  );
};

CustomProgress.propTypes = propTypes;
CustomProgress.defaultProps = defaultProps;

export default withStyles(progressStyles)(CustomProgress);
