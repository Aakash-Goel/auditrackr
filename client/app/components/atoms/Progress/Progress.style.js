import {
  primaryColor,
  infoColor,
  successColor,
  warningColor,
  dangerColor,
  roseColor,
  grayColor,
} from '../../../styles/theme/muiKit';

const progressStyles = {
  root: {
    marginBottom: '20px',
  },
  bar: {},
  primary: {
    backgroundColor: primaryColor,
  },
  warning: {
    backgroundColor: warningColor,
  },
  danger: {
    backgroundColor: dangerColor,
  },
  success: {
    backgroundColor: successColor,
  },
  info: {
    backgroundColor: infoColor,
  },
  rose: {
    backgroundColor: roseColor,
  },
  gray: {
    backgroundColor: grayColor,
  },
  primaryBackground: {
    background: 'rgba(156, 39, 176, 0.2)',
  },
  warningBackground: {
    background: 'rgba(255, 152, 0, 0.2)',
  },
  dangerBackground: {
    background: 'rgba(244, 67, 54, 0.2)',
  },
  successBackground: {
    background: 'rgba(76, 175, 80, 0.2)',
  },
  infoBackground: {
    background: 'rgba(0, 188, 212, 0.2)',
  },
  roseBackground: {
    background: 'rgba(233, 30, 99, 0.2)',
  },
  grayBackground: {
    background: 'rgba(221, 221, 221, 0.2)',
  },
  primaryBufferBackground: {
    background: 'rgba(156, 39, 176, 0.5)',
  },
  warningBufferBackground: {
    background: 'rgba(255, 152, 0, 0.5)',
  },
  dangerBufferBackground: {
    background: 'rgba(244, 67, 54, 0.5)',
  },
  successBufferBackground: {
    background: 'rgba(76, 175, 80, 0.5)',
  },
  infoBufferBackground: {
    background: 'rgba(0, 188, 212, 0.5)',
  },
  roseBufferBackground: {
    background: 'rgba(233, 30, 99, 0.5)',
  },
  grayBufferBackground: {
    background: 'rgba(221, 221, 221, 0.5)',
  },
};

export default progressStyles;
