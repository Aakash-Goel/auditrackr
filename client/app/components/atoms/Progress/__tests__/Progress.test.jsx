import React from 'react';
import TestRenderer from 'react-test-renderer';

import Progress from '../Progress';

describe('<Progress />', () => {
  it('renders correctly', () => {
    const progressWrapper = TestRenderer.create(<Progress />).toJSON();
    expect(progressWrapper).toMatchSnapshot();
  });
});
