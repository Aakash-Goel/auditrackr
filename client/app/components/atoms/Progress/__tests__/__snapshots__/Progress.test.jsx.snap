// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Progress /> renders correctly 1`] = `
<div
  className="MuiLinearProgress-root CustomProgress-root-1 CustomProgress-greyBackground-14 MuiLinearProgress-colorPrimary MuiLinearProgress-indeterminate"
  role="progressbar"
>
  <div
    className="MuiLinearProgress-bar CustomProgress-bar-2 CustomProgress-grey-8 MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
    style={Object {}}
  />
  <div
    className="MuiLinearProgress-bar CustomProgress-bar-2 CustomProgress-grey-8 MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary"
    style={Object {}}
  />
</div>
`;
